### PR TITLE
Fix NSRunLoop cancelPerformSelector arguments

### LINKED ISSUE #1995: cancelPerformSelector... written wrong
Dear WinObjC Engineers,

In NSRunLoop.mm:

- (void)cancelPerformSelector:(const char*)selector target:(NSObject*)target argument:(NSObject*)argument {

is written wrong.

Sincerely,

Ed

### PR DESCRIPTION
**Pull request by aballway**
Change arguments to match documentation, and add to header

Fix #1995

### CODE REVIEW

File: tests/UnitTests/Foundation/NSRunLoopTests.mm [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+@implementation NSRunLoopTest
+
+- (void)shouldNotBeCalled {
+    FAIL();
+}
+
+@end
+
+TEST(NSRunLoop, CancelPerformSelector) {
```
Comment (bbowman) [inline tests/UnitTests/Foundation/NSRunLoopTests.mm]: >NSRunLoop [](start = 5, length = 9)

Any other tests we should add?

**Comment by bbowman** [comment]
:shipit:

File: tests/UnitTests/Foundation/NSRunLoopTests.mm (hunk 1 above)
Comment (aballway) [inline tests/UnitTests/Foundation/NSRunLoopTests.mm]: Probably, but that's a task in and of itself.  Properly testing the whole of NSRunLoop will be nontrivial 😅 though I could add a couple other similar unit tests to this one in this PR if that's what we think would be best.

Comment (rajsesh-msft) [inline tests/UnitTests/Foundation/NSRunLoopTests.mm]: We haven't yet completed the run-loop migration to use CF.  Any additional testing can be covered there.

File: Frameworks/Foundation/NSRunLoop.mm [hunk 2]
```diff
@@ -359,14 +359,14 @@ - (void)performSelector:(SEL)selector target:(id)target argument:(id)argument or
...
-- (void)cancelPerformSelector:(const char*)selector target:(NSObject*)target argument:(NSObject*)argument {
+- (void)cancelPerformSelector:(SEL)selector target:(id)target argument:(id)argument {
     [_orderedLock lock];
     int count = [_orderedPerforms count];
 
     while (--count >= 0) {
         NSOrderedPerform* check = [_orderedPerforms objectAtIndex:count];
 
-        if (strcmp(sel_getName([check selector]), selector) == 0 && [check target] == target && [check argument] == argument) {
+        if ([check selector] == selector && [check target] == target && [check argument] == argument) {
```
Comment (DHowett-MSFT) [inline Frameworks/Foundation/NSRunLoop.mm]: no.

Comment (DHowett-MSFT) [inline Frameworks/Foundation/NSRunLoop.mm]: `sel_isEqual`

Comment (aballway) [inline Frameworks/Foundation/NSRunLoop.mm]: documentation suggests they are equivalent?

**Pull request merged**